Dockerfile

```diff
@@ -0,0 +1,173 @@
+# syntax = docker/dockerfile:1.2
+# WARNING! Use with `docker buildx ...` or `DOCKER_BUILDKIT=1 docker build ...`
+# to enable --mount feature used below.
+
+########################################################################
+# Dockerfile for reproducible build of lavad binary and docker image
+########################################################################
+
+ARG GO_VERSION="1.18.2"
+ARG RUNNER_IMAGE="debian:11-slim"
```

Comment:
@orenl-lava great Dockerfile! Thoughts on using alpine instead of debian-slim for the runner image? We'll get a smaller final image.

Reply:
Yes, that's the plan; I prefer to not delay this PR from getting to main, so deferred optimization until after the merge. 

Currently, our image size is ~256MB, of which the debian slim base (before `apt-get install ...`) is ~81MB; For reference, the alpine base alpine is ~10MB (before `apk add ...`).

Reply:
sounds good, approved